cmd/snap/cmd_run.go

```diff
@@ -44,6 +44,7 @@ type cmdRun struct {
 	Command  string `long:"command" description:"alternative command to run" hidden:"yes"`
 	Hook     string `long:"hook" description:"hook to run" hidden:"yes"`
 	Revision string `short:"r" description:"use a specific snap revision when running hook" default:"unset" hidden:"yes"`
+	HookID   string `short:"i" description:"hook instance's unique ID" hidden:"yes"`
```

Comment:
`-i` is too first-class for something that should only be used internally. We probably don't even need an option for this here, actually. We can set an environment variable before calling snap run and it should persist across the confinement (at least today it does).